### PR TITLE
Config for benchmarking

### PR DESCRIPTION
**Pull request by AnthonySteele**
as per comments
https://github.com/justeat/NLog.StructuredLogging.Json/commit/816128a70b1e9ee64d4c81ae5aac00296ad96c12#r32443869
and https://github.com/justeat/NLog.StructuredLogging.Json/commit/816128a70b1e9ee64d4c81ae5aac00296ad96c12#r32443868

### CODE REVIEW

File: src/Benchmark/nlog.config [hunk 1]
```diff
@@ -23,7 +23,7 @@
       archiveFileName="${archive_dir}\nxlog\{#####}-${level:lowercase=true}.log"
       maxArchiveFiles="36"
       concurrentWrites="false"
-      keepFileOpen="false" />
+      keepFileOpen="true" />
```
Comment (snakefoot) [inline src/Benchmark/nlog.config:26]: Remember also keepFileOpen for the `human_text_file`-target

**Comment by AnthonySteele** [comment]
Benchmarking environment:

```
BenchmarkDotNet=v0.11.3, OS=Windows 10.0.17134.590 (1803/April2018Update/Redstone4)
Intel Core i7-8750H CPU 2.20GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=2.1.500
  [Host]     : .NET Core 2.1.7 (CoreCLR 4.6.27129.04, CoreFX 4.6.27129.04), 64bit RyuJIT
  DefaultJob : .NET Core 2.1.7 (CoreCLR 4.6.27129.04, CoreFX 4.6.27129.04), 64bit RyuJIT
```

Before:

|                        Method |      Mean |     Error |    StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|------------------------------ |----------:|----------:|----------:|------------:|------------:|------------:|--------------------:|
 |                      LogInfo |  6.757 us | 0.0924 us | 0.0819 us |      0.6332 |      0.2136 |      0.0992 |             3.16 KB |
|               LogExtendedInfo | 32.054 us | 0.4275 us | 0.3999 us |      2.1362 |      0.7019 |      0.2747 |             10.7 KB |
 |LogExtendedInfoWithProperties | 36.107 us | 0.5985 us | 0.4998 us |      2.3193 |      0.8545 |      0.1831 |            13.04 KB |
 |                 LogException | 13.072 us | 0.1991 us | 0.1862 us |      1.2665 |      0.4272 |      0.1831 |              6.4 KB |
 |         LogExtendedException | 40.531 us | 0.8479 us | 1.0093 us |      2.6855 |      0.9155 |      0.1221 |            14.56 KB |
 |   LogExceptionWithProperties | 45.288 us | 0.8398 us | 0.7855 us |      3.1128 |      1.1597 |      0.1831 |            16.82 KB |


After:

|                        Method |     Mean |     Error |    StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|------------------------------ |---------:|----------:|----------:|------------:|------------:|------------:|--------------------:|
 |                      LogInfo | 4.806 ms | 0.2009 ms | 0.5765 ms |           - |           - |           - |            35.13 KB |
 |              LogExtendedInfo | 4.145 ms | 0.0409 ms | 0.0363 ms |      7.8125 |           - |           - |            42.58 KB |
 |LogExtendedInfoWithProperties | 4.707 ms | 0.1707 ms | 0.4979 ms |           - |           - |           - |            46.97 KB |
 |                 LogException | 3.978 ms | 0.0335 ms | 0.0297 ms |      7.8125 |           - |           - |            38.46 KB |
 |         LogExtendedException | 4.136 ms | 0.0812 ms | 0.0935 ms |      7.8125 |           - |           - |            46.44 KB |
  |  LogExceptionWithProperties | 4.101 ms | 0.0779 ms | 0.0800 ms |      7.8125 |           - |           - |            48.52 KB |

Yes, this is worse.

The point of this PR is not to produce the best performance numbers by tweaking the benchmark, it is to produce a stable and accurate benchmark so that the code can be tweaked in light of the benchmark results.

**Comment by snakefoot** [comment]
> The point of this PR is not to produce the best performance numbers by tweaking the benchmark

I'm guessing you are trying to benchmark your own code. So if forcing NLog-engine to run with maximum overhead, then your own code will be dwarfed by the NLog-engine.

**Comment by AnthonySteele** [comment]
Yes, that's true.

**Comment by snakefoot** [comment]
One might even consider using the Null-target with `formatMessage=true` instead of using File-target: https://github.com/NLog/NLog/wiki/Null-target

**Comment by AnthonySteele** [comment]
closed in favour of #141

**Pull request closed without merging**